### ISSUE TITLE
Calibrating multiple industries

### ISSUE THREAD

**Issue by rickecon** [EAPD-DRB/OG-ZAF#29]
We want to calibrate multiple industries in OG-ZAF.

**What industries to calibrate?**
The first step is to choose what industries to calibrate. I recommend the following, which come from the South African Reserve Bank [Quarterly Bulletin](https://www.resbank.co.za/en/home/publications/publication-detail-pages/quarterly-bulletins/quarterly-bulletin-publications/2022/full-quarterly-bulletin---no-306---december-2022):
* Primary sector: Agriculture, forestry, and fishing + Mining and quarrying
* Secondary sector (this sector must be the `M`th industry because it will be the industry that produces goods that can be used as capital): Manufacturing + Electricity, gas and water + Construction (contractors)
* Tertiary sector: Wholesale and retail trade, catering and accommodation + Transport, storage and communication + Finance, insurance, real estate and business services + Government services + Other

Another industry we could include is the informal sector, which is a production sector that does not pay corporate income taxes.

**What parameters and what data are needed to calibrate these 3 or 4 industries?**
Let `m` be the index of the `m`th industry with the total number of industries being `M`. Let `i` be the index of the `i`th consumption good with the total number of consumption goods being `I`. The theory for the parameters listed below is in the OG-Core documentation for the theory of [the Firm](https://pslmodels.github.io/OG-Core/content/theory/firms.html) and [Households](https://pslmodels.github.io/OG-Core/content/theory/households.html).
* Number of industries `M`: Number of industries is set in `ogzaf_default_parameters.json` as the parameter scalar `M`.
* Number of consumption goods `I`: The choice of the number of consumption goods depends on the research questions you want to answer and the quality of the input output data you have for the country. The cheapest calibration is to assume that each production industry is a good that households consume. In this case `I = M`. But `I` does not have to equal `M` in general.
* Total factor productivity `Z_m`: TFP is represented in the theory as `Z_m` and its value is given by the list of lists parameter `Z` in `ogzaf_default_parameters.json`. For example, if `M=3`, a corresponding value for `Z` is `Z = [[1.0., 1.2, 0.9]]`. The `Z` value for a given industry is estimated by a standard log regression of industry output on industry capital stock and industry labor hours. Because the model units are endogenous, the important thing about calibrating the `Z` value for each industry is getting its relative size right. As such, it makes sense to normalize one of the industries TFP to 1.0 and set the other TPF values so that the relative size of the industry matches the output data.
* Constant elasticity of substitution `epsilon`. A Cobb-Douglas specification for `M=3` industries would be `epsilon = [1.0, 1.0, 1.0]`.
* For now, let's exclude public capital `K_g` from the production function using the `gamma_g` parameter. If the number of industries is `M=3`, the correct specification would be `gamma_g = [0.0, 0.0, 0.0]`. This parameter allows us to account for government infrastructure investment that goes into the production function of each industry. Although this is sufficient, we should set `initial_Kg_ratio` to zero as well.
* Capital share of income. In the documentation of the theory, this parameter is `gamma_m`. In the `ogzaf_default_parameters.json`, the parameter is `gamma` and is a list of scalars strictly between 0 and 1. For example, if `M=3` a valid specification would be `gamma = [0.2, 0.5, 0.3]`. Because labor cost data for industries is usually more accurate than capital cost data, we calibrate the capital share of income by calculating the labor share of income. Then the capital share is 1 - labor share. Labor share = total labor compensation / output, where total labor compensation is employee compensation plus proprietor compensation.
* Input-output matrix `io_matrix`: This parameter maps production industry output to consumption categories. The easiest calibration for a model with `M=3` and `I=3` is a 3 x 3 identity matrix input into `ogzaf_default_parameters.json` in the following way.
```
"io_matrix": [
                    [1.0, 0.0, 0.0],
                    [0.0, 1.0, 0.0],
                    [0.0, 0.0, 1.0]
                ],
```
* Consumption share in consumption aggregator function `alpha_c`: In the documentation, the consumption share is defined as `alpha_i` (see equation [10](https://pslmodels.github.io/OG-Core/content/theory/households.html#equation-eqhhcompcons)). In OG-Core, the parameter is `alpha_c`. A valid calibration for a model with `I=3` consumption goods would be `alpha_c = [0.3, 0.4, 0.3]`.
* Corporate income tax rate `cit_rate`
* Consumption tax rate `tau_c`

cc: @SeaCelo @sarvonz @jdebacker

**Comment by rickecon** [comment]
I found this South African central bank paper (Pain, et al, 2020) estimating TFP in South Africa by industry. It looks very helpful in that they estimate TFP growth rates for South African industries from 2009- 2020. However, it is not clear to me how to back out the Z values rather than the growth rates (log difference).

* Julius Pain, Mpho Rapapali and Daan Steenkamp, "[Industry TFP estimates for South Africa](https://econpapers.repec.org/scripts/redir.pf?u=https%3A%2F%2Fwww.resbank.co.za%2Fcontent%2Fdam%2Fsarb%2Fpublications%2Foccasional-bulletin-of-economic-notes%2F2020%2F10412%2FOBEN%25202002%2520%28Industry%2520TFP%2520estimates%2520for%2520South%2520Africa%29%2520-%2520November%25202020.pdf;h=repec:rbz:oboens:10412)", Chapter 5 in *Occasional Bulletin of Economic Notes*, OBEN/20/02, South African Reserve Bank, Nov. 2020.

Furthermore, I can't tell what method Pain, et al (2020) are using to account for the various types of endogeneity described in Beveren (2012; see below for full reference). But I might just need to read Pain, et al (2020) more carefully.

* Ilke Van Beveren, "[Total Factor Productivity Estimation: A Practical Review](https://onlinelibrary.wiley.com/doi/full/10.1111/j.1467-6419.2010.00631.x?casa_token=P2GBG5wKAqQAAAAA%3A6zBROGdCkmn-4yFWAM3xHa5YLSuon-kHuij_zjcFY_Vb7MKcOel8dNRGoJkeBjpRIsx1hBL0SXEutuk)," *Journal of Economic Surveys*, vol. 26, issue 1, pp. 98-128, (Feb. 2012). [Get non-gated working paper version [here](https://feb.kuleuven.be/drc/licos/publications/dp/dp182.pdf).]

cc: @SeaCelo @sarvonz @jdebacker

**Comment by SeaCelo** [comment]
On TFP, just to add that a possible alternative is the South African Reserve Bank [Quarterly Bulletin](https://www.resbank.co.za/en/home/publications/publication-detail-pages/quarterly-bulletins/quarterly-bulletin-publications/2022/full-quarterly-bulletin---no-306---december-2022) that you shared. It has most (all?) the data needed to compute the TFP or an approximation (Output, labor, capital) by industry. I'm not clear on how to convert these to the `Z_m` variable and the documentation doesn't make clear what the units/base should be.

On Capital share of income `gamma_m`, the OECD has [Employee compensation by activity](https://data.oecd.org/earnwage/employee-compensation-by-activity.htm) as a share of total value added (Agriculture, industry, services). 
For 2020: 

Sector | Share of gross VA | Share of Labor compensation
-- | -- | --
Services | 38.7 | 0.72
Agriculture | 0.8 | 0.01
Industry | 14.5 | 0.27
Total | 54.0 | 1.00

@rickecon @sarvonz @jdebacker

**Comment by sarvonz** [comment]
We have done some TFP calculations. Given the industry-level data on value added, fixed capital stock, people employed and employee compensation from the South African Reserve Bank, we used `Y=AF(K,L)`, with `F(K,L)=(alpha*(K^rho)+(1-alpha)*(L^rho))^(1/rho)`, to back out A. This specification of `F(K,L)` was estimated in the SARB paper Pain et al (2020). Rho is `(sigma-1)/sigma`, and sigma is set as 0.9 following the paper.

Marcelo reached out to SARB and they were very helpful with providing the data.

For 2021:

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/HOI~1.CHE/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/HOI~1.CHE/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



  | TFP (=GVA/F(K,L)) | TFP (Secondary=1)
-- | -- | --
Primary | 396.80 | 0.17
Secondary | 2377.11 | 1.00
Tertiary | 6604.17 | 2.78



</body>

</html>

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/HOI~1.CHE/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/HOI~1.CHE/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">



  | TFP (=GVA/F(K,L)) | TFP (Secondary, excl. energy=1)
-- | -- | --
Primary | 396.80 | 0.11
Secondary (excl. energy) | 3474.10 | 1.00
Tertiary | 6604.17 | 1.90
Energy | 551.63 | 0.16

</body>

</html>

@rickecon @SeaCelo @jdebacker

**Comment by sarvonz** [comment]
Capital share of income, by industry (2021)

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/HOI~1.CHE/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/HOI~1.CHE/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



  | Capital share of income
-- | --
Agriculture | 0.67
Mining and quarrying | 0.67
Manufacturing | 0.52
Electricity, gas and water supply | 0.68
Construction | 0.36
Wholesale, retail and motor  trade; hotels and restaurnts | 0.51
Transport, storage and communication | 0.57
Financial intermediation, insurance, real   estate and business services | 0.60
Community, social and personal servcies | 0.24
  |  
  | Capital share of income
primary sector | 0.67
secondary sector | 0.53
secondary sector, without   energy | 0.50
tertiary sector | 0.45



</body>

</html>

@SeaCelo @rickecon @jdebacker

**Comment by SeaCelo** [comment]
Closed with #31
